### PR TITLE
ZCS-11201 Fix counter parameter sanitized issue

### PR DESCRIPTION
**Pull request by deepakzimb**
Problem: "counter" parameter can accept HTML entity from URL and it needs to be sanitized. 

Solution: "counter" parameter sanitized with the method and after sanitization, it is working fine.

### CODE REVIEW

**Review by preshitaagrawal — APPROVED** [review]

File: WebRoot/WEB-INF/tags/message/displayMessage.tag [hunk 1]
```diff
@@ -342,7 +342,7 @@
         </tr>
     </c:if>
     <tr>
-        <td id="iframeBody${counter}" class=MsgBody valign='top' colspan="${needExtraCol ? 1 : 2}">
+        <td id="iframeBody${fn:escapeXml(counter)}" class=MsgBody valign='top' colspan="${needExtraCol ? 1 : 2}">
             <app:body message="${message}" body="${body}" theBody="${body.isTextHtml ? zm:stripHtmlComments(theBody) : theBody}" mailbox="${mailbox}" counter="${counter}"/>
```
Comment (silentsakky) [inline WebRoot/WEB-INF/tags/message/displayMessage.tag:346]: we are using counter here as well, should we need to escape it?

Comment (deepakzimb) [reply WebRoot/WEB-INF/tags/message/displayMessage.tag:346]: @silentsakky   This is used for the body tag and the leaf node will be the body.tag file, we can add escape in body.tag file for safe side. Currently, it is not affecting but we can add escape in body.tag file, your suggestion are most welcome.

Comment (silentsakky) [reply WebRoot/WEB-INF/tags/message/displayMessage.tag:346]: `https://zdev-vm149.eng.zimbra.com/h/message?id=857&print=true&counter=hhhhhhhh%22%3E%3Cimg%20src=x%20onerror=alert(document.domain)%3E`
above endpoint is also vulnerable after the fix

Comment (silentsakky) [reply WebRoot/WEB-INF/tags/message/displayMessage.tag:346]: `https://zdev-vm149.eng.zimbra.com/h/printconversations?id=-858&print=true&counter=hhhhhhhh%22%3E%3Cimg%20src=x%20onerror=alert(document.domain)%3E`
`https://zdev-vm149.eng.zimbra.com/h/printmessage?id=858&print=true&counter=hhhhhhhh%22%3E%3Cimg%20src=x%20onerror=alert(document.domain)%3E`

Comment (deepakzimb) [reply WebRoot/WEB-INF/tags/message/displayMessage.tag:346]: Fixed in body and print file.

**Review by silentsakky — APPROVED** [review]

**Review by zimsuchitgupta — APPROVED** [review]

**Pull request merged**